### DISCUSSION TITLE
1BRC Dart Single Core

### DISCUSSION THREAD

**Discussion by HankG** [gunnarmorling/1brc#241] [Show and tell]
I tried my hand at doing this in Dart, my preferred language. On my machine the standard Java implementation ran in about 2 minutes 14 seconds. 

The naive Dart implementation not using immutable objects ran in 6 minutes 50 seconds, or about 300% slower:
https://gitlab.com/HankG/1brc/-/blob/6742515ff1c5702c0534ff2b72f418a2d8adc826/bin/onebrc.dart

The as many stops out as I could think of version runs in 2 minutes 43 seconds, so about 22% slower:
https://gitlab.com/HankG/1brc/-/blob/6742515ff1c5702c0534ff2b72f418a2d8adc826/bin/onebrc.dart

A concurrency example using Dart Isolates I started dabbling with but didn't have anything worth showing for it. The nature of Isolates/Workers model over a more traditional multithreading model makes that more difficult for me to get my head around how I'd get performance out of it.

**Comment by sanathusk** [comment]
@HankG I think we can optimize this further, i have a [PR](https://gitlab.com/HankG/1brc/-/merge_requests/1) for the same please merge the same if you agree with improvements https://gitlab.com/HankG/1brc/-/merge_requests/1

This implementation completes under 2 seconds (~1.17 seconds) on my machine
real    0m1.171s
user    0m0.239s
sys     0m0.044s

**Reply by AlexanderYastrebov** [reply]
I think using sorted map may slowdown things but the code should be certainly pre-compiled like you suggest :+1:
